python/lib/sift_py/ingestion/config/yaml/load.py

```diff
@@ -275,11 +335,39 @@ def _validate_bit_field_element(val: Any):
 def _validate_rule(val: Any):
     rule = cast(Dict[Any, Any], val)
 
+    namespace = rule.get("namespace")
+    if namespace is not None and not isinstance(namespace, str):
+        raise YamlConfigError._invalid_property(
+            namespace,
+            "- namespace",
+            "str",
+            ["rules"],
+        )
+
     name = rule.get("name")
 
     if not isinstance(name, str):
         raise YamlConfigError._invalid_property(name, "- name", "str", ["rules"])
 
+    channel_references = rule.get("channel_references")
+
+    if namespace or (channel_references is not None):
+        if not isinstance(channel_references, list):
+            raise YamlConfigError._invalid_property(
+                channel_references,
+                "- channel_references",
+                f"List[Dict[str, {_type_fqn(ChannelConfigYamlSpec)}]]",
+                ["rules"],
+            )
+
+        for channel_reference in cast(List[Any], channel_references):
+            _validate_channel_reference(channel_reference)
+
+    if namespace:
+        # TODO: Probably should inform the user that the below keys
```

Comment:
Hopefully this makes sense -- I was thinking that if a user is trying to reference a rule in a namespace it would be nice to clearly tell them that the below keys should not be included in the top level yaml.